File: recyclerView/app/src/main/java/com/example/recyclerview/MainAdapter.java
package com.example.recyclerview;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.view.menu.MenuView;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;


public class MainAdapter extends RecyclerView.Adapter<MainAdapter.CustomViewHolder> {

    //리스트 뷰의 아이템들을 담을 배열을 선언해준 부분이다.
    private ArrayList<MainData> arrayList;

    public MainAdapter(ArrayList<MainData> arrayList) {
        this.arrayList = arrayList;
    }

    @NonNull
    @Override
    //리스트뷰가 처음으로 생성이 될떄 동작
    public MainAdapter.CustomViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_list, parent,false);
        CustomViewHolder holder = new CustomViewHolder(view);

        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull MainAdapter.CustomViewHolder holder, int position) {
        //실제 추가될때!
        holder.iv_profile.setImageResource(arrayList.get(position).getIv_profile());
        holder.tv_name.setText(arrayList.get(position).getTv_name());
        holder.tv_content.setText(arrayList.get(position).getTv_content());

        holder.itemView.setTag(position);
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //object 형태를 string 형태로 벼환해서 받음
                String curName = holder.tv_name.getText().toString();
                Toast.makeText(v.getContext(), curName, Toast.LENGTH_SHORT).show();

            }
        });

        holder.itemView.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {
                remove(holder.getAdapterPosition());

                return true;
            }
        });
    }

    @Override
    public int getItemCount() {
        return (null != arrayList ? arrayList.size() : 0);
    }

    //길게 누르면 삭제할 수 있는 부분!
    public void remove(int position){
        try{
            //리스트를 지운후에 새로고침을 해야 실제로 보여진다!
            arrayList.remove(position);
            notifyItemRemoved(position); //새로고침이라는 뜻이다.
        } catch(IndexOutOfBoundsException ex) {
            ex.printStackTrace(); //예외 상황이 벌어졌을때 강제 실행
        }

    }

    public class CustomViewHolder extends RecyclerView.ViewHolder {

        protected ImageView iv_profile;
        protected TextView tv_name;
        protected TextView tv_content;

        public CustomViewHolder(@NonNull View itemView) {
            super(itemView);
            this.iv_profile = (ImageView) itemView.findViewById(R.id.iv_profile);
            this.tv_name = (TextView) itemView.findViewById(R.id.tv_name);
            this.tv_content = (TextView) itemView.findViewById(R.id.tv_content);
        }
    }
}

File: fixed_habit/app/src/main/java/com/inwooshin/fixed_habit/MainAdapter.kt
package com.inwooshin.fixed_habit

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import java.util.*


class MainAdapter(arrayList: ArrayList<MainData>?) :
    RecyclerView.Adapter<MainAdapter.CustomViewHolder>() {
    //리스트 뷰의 아이템들을 담을 배열을 선언해준 부분이다.
    private val arrayList: ArrayList<MainData>?

    //리스트뷰가 처음으로 생성이 될떄 동작
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): CustomViewHolder {
        val view: View =
            LayoutInflater.from(parent.context).inflate(R.layout.dynamic_add_list, parent, false)
        return CustomViewHolder(view)
    }

    override fun onBindViewHolder(holder: CustomViewHolder, position: Int) {
        //실제 추가될때!
        holder.iv_profile.setImageResource(arrayList!![position].getIv_profile())
        holder.tv_title.setText(arrayList[position].getTv_title())
        holder.tv_time.setText(arrayList[position].getTv_name())
        holder.tv_content.setText(arrayList[position].getTv_content())
        holder.itemView.tag = position
        holder.itemView.setOnClickListener { v -> //object 형태를 string 형태로 벼환해서 받음
            val curName = holder.tv_title.text.toString()
            var stateMessage : String = holder.tv_title.text.toString() + "를 " + holder.tv_time.text.toString() + "에 수행할 예정입니다."
            Toast.makeText(v.context, stateMessage, Toast.LENGTH_SHORT).show()
        }
        holder.itemView.setOnLongClickListener {
            remove(holder.adapterPosition)
            true
        }
    }

    override fun getItemCount(): Int {
        return arrayList?.size ?: 0
    }

    //길게 누르면 삭제할 수 있는 부분!
    fun remove(position: Int) {
        try {
            //리스트를 지운후에 새로고침을 해야 실제로 보여진다!
            arrayList!!.removeAt(position)
            notifyItemRemoved(position) //새로고침이라는 뜻이다.
        } catch (ex: IndexOutOfBoundsException) {
            ex.printStackTrace() //예외 상황이 벌어졌을때 강제 실행
        }
    }

    inner class CustomViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var iv_profile: ImageView
        var tv_title: TextView
        var tv_time: TextView
        var tv_content: TextView

        init {
            iv_profile = itemView.findViewById<View>(R.id.add_list_img) as ImageView
            tv_title = itemView.findViewById<View>(R.id.add_list_text) as TextView
            tv_time = itemView.findViewById<View>(R.id.add_list_time) as TextView
            tv_content = itemView.findViewById<View>(R.id.add_list_week) as TextView
        }
    }

    init {
        this.arrayList = arrayList
    }
}
